src/fast-path.js

```diff
@@ -68,7 +68,9 @@ function getNodeHelper(path, count) {
 
   for (var i = s.length - 1; i >= 0; i -= 2) {
     var value = s[i];
-    if (n.Node.check(value) && --count < 0) {
+    // Temp: This can be removed when `ast-types` knows that TSNodes are Nodes.
+    var isTsNode = value && value.type && value.type.indexOf('TS') === 0;
```

Comment:
Had to use a bit of a workaround here because `ast-types` doesn't know about the `TS*` nodes. Didn't want to step on people like @Pajn's toes with that side of things.